### PR TITLE
Froniter Oven and Medical Assembler Port

### PR DESCRIPTION
**Pull request by Dunrab**
## About the PR
Oh boy here we go, this PR ports Frontiers awesome Medical Assembler, special medical pens that can be made by chem in medical assembler and Frontiers Oven that can be used to cook specific sets of foods.

Original PRs from Froniter, credit too the people who worked to get all of this working.
Frontier PR that added the medical assembler & oven: [Frontier #1935](https://github.com/new-frontiers-14/frontier-station-14/pull/1935)
Frontier PR that makes the guidebook for the assembler auto-generated: [Frontier #2331](https://github.com/new-frontiers-14/frontier-station-14/pull/2331)

TODO:
- [x] Link original PRs to give proper credit to the people who created these lovely machines
- [x] Test the hell out of this. I have not tested this enough yet.
- [x] Need to fix guidebook entries, they are pretty broken
- [x] Recipies seem to work fine, need to test research & add the basic empty pen to the medifab or research not sure which would be better

## Why / Balance
The medical assembler is a nice tool for chem to produce medical pens to give out to salvage, doctors, security or who ever. It is a great RP opportunity on frontier and I believe it will benefit RP here nicely.

The oven will be used in a future PR so specific recipes can be used in it.

The medical assembler and empty medipens are t3 biomedical research and the oven range is t2 civilian.

I increased the price per empty medipen but we can increase it further if desired.

## Technical details
I ported both machines since they share code and figured why not knock two birds with one stone since both will provide small content updates for each department.

## Media
<details> <summary><h3>Click to show</h3></summary> <p>

https://github.com/user-attachments/assets/cde94baf-d48c-482b-8922-666e3393e0d8

</p></details>

## Requirements

- [x] I have tested all added content and changes.
- [x] I have added media to this PR or it does not require an ingame showcase.

## Licensing:

- [ ] I confirm that I am the creator of the content in this PR, and allow licensing it under the following license(s), or that the original author has given me permission to do so:
  - [X] AGPL (https://github.com/Floof-Station/Panta-Rhei/blob/master/LICENSE-AGPLv3.txt) 
  - [ ] MIT (https://github.com/Floof-Station/Panta-Rhei/blob/master/LICENSE-MIT.txt)
    

## Breaking changes

**Changelog**

:cl:
- add: Frontier Medical Assembler & Oven have been ported and be found in t3 medical research and t2 civilian respectively.
- add: Once researched, medical can print empty medical pens that can be used in the medical assembler.
- add: The oven can be used for most recipes currently and functions as a stove & electric grill combined.

### CODE REVIEW

**Review by Mnemotechnician — COMMENTED** [review]
The fact that this changes the microwave system instead of doing literally anything else is awful

File: Resources/Prototypes/_NF/Entities/Structures/Machines/mirowave_recipe_assemblers.yml [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+    sprite: _NF/Structures/Machines/oven_syndie.rsi
+  - type: Machine
+    board: SyndicateElectricRangeMachineCircuitboard
+  - type: FoodRecipeProvider
+    providedRecipes:
+    - RecipeBaguetteSword
+    - RecipeThrowingCroissant
+
+- type: entity
+  id: KitchenAssembler
```
Comment (SyaoranFox) [inline Resources/Prototypes/_NF/Entities/Structures/Machines/mirowave_recipe_assemblers.yml:145]: What will be made with this? I cant find any recipes as part of your PR for the machine.

File: Resources/Prototypes/_NF/Entities/Structures/Machines/mirowave_recipe_assemblers.yml [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+  components:
+  - type: Microwave
+    capacity: 10
+    canMicrowaveIdsSafely: false
+    explosionChance: 0
+    lightningChance: 0
+    beginCookingSound: /Audio/_NF/Machines/egg-timer.ogg
+    foodDoneSound: /Audio/Machines/ding.ogg
+    loopingSound: /Audio/_NF/Effects/silence.ogg # need to change this sound
+    validRecipeTypes: Oven
```
Comment (SyaoranFox) [inline Resources/Prototypes/_NF/Entities/Structures/Machines/mirowave_recipe_assemblers.yml:16]: Is it intended to be able to cook microwave recipes in the oven at the moment? Will there be a time when you can't, or does the component "microwave" mean it always will?

**Review by SyaoranFox — APPROVED** [review]

File: Resources/Prototypes/_NF/Entities/Structures/Machines/mirowave_recipe_assemblers.yml (hunk 2 above)
Comment (Dunrab) [reply Resources/Prototypes/_NF/Entities/Structures/Machines/mirowave_recipe_assemblers.yml:16]: So the range being able to cook microwave recipes is temporary, i felt it would to be silly to add it and it have no use at all, we can pull it research though if we want to remove that for now

File: Resources/Prototypes/_NF/Entities/Structures/Machines/mirowave_recipe_assemblers.yml (hunk 1 above)
Comment (Dunrab) [reply Resources/Prototypes/_NF/Entities/Structures/Machines/mirowave_recipe_assemblers.yml:145]: This is not included in research, so this is admin only for now till we decide if we even want to keep it and or use it, maybe we can use this for machine for that one bounty thing that chefs could do for cash with bultertron?

Comment (SyaoranFox) [reply Resources/Prototypes/_NF/Entities/Structures/Machines/mirowave_recipe_assemblers.yml:145]: That's fine. We should keep it so they can assemble things. I am just unsure what that would be. Like, why have an assembler, when we have the lathes (both service and medical)?

Comment (Dunrab) [reply Resources/Prototypes/_NF/Entities/Structures/Machines/mirowave_recipe_assemblers.yml:145]: If im remembering right on Frontier, this was not used in that way, it was used for a few very specific food items, it has been a while though so that may have changed with time

**Comment by SyaoranFox** [comment]
> The fact that this changes the microwave system instead of doing literally anything else is awful

What do you need to see for this to be okay to merge in? Or is it okay enough for now?

**Comment by Dunrab** [comment]
marking as dnm pending wizdens microwave refactor or potential microwave removal

File: Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs [hunk 3]
```diff
@@ -478,6 +501,7 @@ public void UpdateUserInterfaceState(EntityUid uid, MicrowaveComponent component
                 component.CurrentCookTimerTime,
                 component.CurrentCookTimeEnd
             ));
+            */
```
Comment (Mnemotechnician) [inline Content.Server/Kitchen/EntitySystems/MicrowaveSystem.cs:504]: Would make more sense to edit the original statmenet instead of duplicatin it

File: Content.Server/Content.Server.csproj [hunk 4]
```diff
@@ -32,6 +32,7 @@
     <Folder Include="_Euphoria\" />
     <Folder Include="_Floof\Lewd\" />
     <Folder Include="_Goobstation\Power\" />
+    <Folder Include="_NF\Acess\Systems\" />
```
Comment (Mnemotechnician) [inline Content.Server/Content.Server.csproj]: Autogenerated, not necessary
```suggestion
```

File: Content.Shared/Placeable/ItemPlacerSystem.cs [hunk 5]
```diff
@@ -41,7 +41,6 @@ private void OnStartCollide(EntityUid uid, ItemPlacerComponent comp, ref StartCo
 
         if (comp.MaxEntities > 0 && count >= (comp.MaxEntities - 1))
         {
-            // Don't let any more items be placed if it's reached its limit.
```
Comment (Mnemotechnician) [inline Content.Shared/Placeable/ItemPlacerSystem.cs:44]: I dont think this is supposed to be removed

**Review by Mnemotechnician — APPROVED** [review]
This is probably more or less fine, just a few nitpicks

**Comment by Mnemotechnician** [comment]
Eh, fuck it
